#! /bin/sh /usr/share/dpatch/dpatch-run
## mscodec_null_name.dpatch by Samuel Mimram <samuel.mimram@ens-lyon.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Check if the name arguments are NULL (if it's the case, we should return NULL).

@DPATCH@
diff -urNad linphone-1.0.0pre4/mediastreamer/mscodec.c /tmp/dpep.w47mIR/linphone-1.0.0pre4/mediastreamer/mscodec.c
--- linphone-1.0.0pre4/mediastreamer/mscodec.c	2003-06-08 08:17:54.000000000 +0200
+++ /tmp/dpep.w47mIR/linphone-1.0.0pre4/mediastreamer/mscodec.c	2004-10-31 16:51:30.000000000 +0100
@@ -69,7 +69,7 @@
 {
 	GList *elem=filter_list;
 	MSFilterInfo *info;
-	while (elem!=NULL)
+	while (elem!=NULL && name!=NULL)
 	{
 		info=(MSFilterInfo *)elem->data;
 		if ( (info->type==MS_FILTER_AUDIO_CODEC) ){
@@ -87,7 +87,7 @@
 {
 	GList *elem=filter_list;
 	MSFilterInfo *info;
-	while (elem!=NULL)
+	while (elem!=NULL && name!=NULL)
 	{
 		info=(MSFilterInfo *)elem->data;
 		if ( (info->type==MS_FILTER_VIDEO_CODEC) ){
@@ -122,7 +122,7 @@
 {
 	GList *elem=filter_list;
 	MSFilterInfo *info;
-	while (elem!=NULL)
+	while (elem!=NULL && name!=NULL)
 	{
 		info=(MSFilterInfo *)elem->data;
 		if ((info->type==MS_FILTER_AUDIO_CODEC) || (info->type==MS_FILTER_VIDEO_CODEC)){
@@ -140,7 +140,7 @@
 {
 	GList *elem=filter_list;
 	MSFilterInfo *info;
-	while (elem!=NULL)
+	while (elem!=NULL && name!=NULL)
 	{
 		info=(MSFilterInfo *)elem->data;
 		if ((info->type==MS_FILTER_AUDIO_CODEC) || (info->type==MS_FILTER_VIDEO_CODEC)){
